python/lsst/ts/mtaos/model.py

```diff
@@ -1021,6 +1026,85 @@ def _generate_pipetask_command(
 
         return run_pipetask_cmd
 
+    async def get_image_info(self, visit_id):
+        """Get image information from the butler.
+
+        Parameters
+        ----------
+        visit_id : `int`
+            Visit id of the image.
+
+        Returns
+        -------
+        filter : `str`
+            Filter of the image.
+        rotation_angle : `float`
+            Rotation angle of the image.
+        elevation : `float`
+            Elevation of the image.
+
+        Raises
+        ------
+        ValueError
+            If the visit id is not found in the butler.
+        """
+        butler = dafButler.Butler(self.data_path)
+        refs = butler.query_datasets("raw", where=f"visit={visit_id}")
+
+        if len(refs) == 0:
+            raise ValueError(
+                f"Visit {visit_id} has no associated images in the butler."
+            )
+
+        image = butler.get(refs[0])
+        filter = image.getFilter().bandLabel
+        rotation_angle = image.getMetadata().get("ROTPA")
+        elevation = (
+            image.getMetadata().get("ELSTART") + image.getMetadata().get("ELEND")
+        ) / 2
+
+        return filter, rotation_angle, elevation
+
+    async def should_apply_corrections(self, visit_id: int) -> float:
```

Comment:
This is a bit weird, the name of the method, "should" implies that it will return a boolean True/False, but it is actually returning a float gain. Please consider renaming this method, maybe something like `get_correction_gain`?